handler/src/test/java/io/netty/handler/ssl/RenegotiateTest.java

```diff
@@ -0,0 +1,132 @@
+
+import io.netty.bootstrap.Bootstrap;
+import io.netty.bootstrap.ServerBootstrap;
+import io.netty.channel.Channel;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.channel.ChannelInitializer;
+import io.netty.channel.EventLoopGroup;
+import io.netty.channel.local.LocalAddress;
+import io.netty.channel.local.LocalChannel;
+import io.netty.channel.local.LocalEventLoopGroup;
+import io.netty.channel.local.LocalServerChannel;
+import io.netty.handler.ssl.util.InsecureTrustManagerFactory;
+import io.netty.handler.ssl.util.SelfSignedCertificate;
+import io.netty.util.ReferenceCountUtil;
+import io.netty.util.concurrent.Future;
+import io.netty.util.concurrent.FutureListener;
+import org.junit.Test;
+
+import java.util.concurrent.CountDownLatch;
+
+public abstract class RenegotiateTest {
+
+    @Test(timeout = 5000)
+    public void testRenegotiateServer() throws Exception {
+        final CountDownLatch latch = new CountDownLatch(2);
+        SelfSignedCertificate cert = new SelfSignedCertificate();
+        EventLoopGroup group = new LocalEventLoopGroup();
+        try {
+            final SslContext context = SslContextBuilder.forServer(cert.key(), cert.cert())
+                    .sslProvider(serverSslProvider()).build();
+            ServerBootstrap sb = new ServerBootstrap();
+            sb.group(group).channel(LocalServerChannel.class)
+                    .childHandler(new ChannelInitializer<Channel>() {
+                        @Override
+                        protected void initChannel(Channel ch) throws Exception {
+                            ch.pipeline().addLast(context.newHandler(ch.alloc()));
+                            ch.pipeline().addLast(new ChannelInboundHandlerAdapter() {
+                                private boolean renegotiate;
+
+                                @Override
+                                public void channelRead(ChannelHandlerContext ctx, Object msg) throws Exception {
+                                    ReferenceCountUtil.release(msg);
+                                }
+
+                                @Override
+                                public void userEventTriggered(
+                                        final ChannelHandlerContext ctx, Object evt) throws Exception {
+                                    if (!renegotiate && evt instanceof SslHandshakeCompletionEvent) {
+                                        SslHandshakeCompletionEvent event = (SslHandshakeCompletionEvent) evt;
+
+                                        if (event.isSuccess()) {
+                                            final SslHandler handler = ctx.pipeline().get(SslHandler.class);
+
+                                            renegotiate = true;
+                                            handler.engine().setEnabledCipherSuites(
+                                                    new String[]{"SSL_RSA_WITH_3DES_EDE_CBC_SHA"});
```

Comment:
add a comment as to why this specific cipher is needed, or why it must be limited to 1? Can we make use of [SupportedCipherSuiteFilter](https://github.com/netty/netty/blob/master/handler/src/main/java/io/netty/handler/ssl/SupportedCipherSuiteFilter.java)